systemd/moodle-docker.service

```diff
@@ -0,0 +1,21 @@
+[Unit]
+Description=Docker based Moodle Deployment
+Documentation=https://github.com/SecurityCompass/moodle-docker
+After=docker.service
+Requires=docker.service
+
+[Service]
+WorkingDirectory=/etc/moodle-docker
+Type=oneshot
```

Comment:
`docker-compose` doesn't stay running, the actual containers are managed by `docker`

Reply:
If you remove the --detach it will stay running, that way the logs show up properly with journalctl.

Reply:
`--detach` is safer in prod because it saves on resources.. plus we should have a proper logging solution vs `docker` -> `docker-compose` -> `journalctl`

Some more notes here:
https://github.com/docker/compose/issues/4266#issuecomment-308324623

Reply:
FYI, I would much rather we use the 'syslog' docker logging driver and leave compose detached.  This is what I'm doing for SDE, and instead of dumping logs into the compose stdout, it will send them to syslog with an appropriate container tag.  Let me know if you have questions about this.

Reply:
What resources would it be using exactly? The example posted before that comment is the correct way to do this IMO. The proper logging solution is to use this as a simple service and not oneshot. Then the logs work exactly as you stated, docker > docker-compose > journalctl.

What's the behaviour with stopping a container in a oneshot service? once docker-compose exits the oneshot service will report it's exited successfully. This could leave you in a weird state where the container is running/crashed but systemd is not aware of this.

Also what happens when docker-compose exits but the container stops right after due to an error or some other reason? In this case systemd wont print any errors and the oneshot service will report success even though the container is not running.

Another reason to make this a simple service would be then we can use the Restart and related config fields to ensure the container is brought back up if it exits for some reason.

Reply:
Discussed further with @hrpatel, I think we are good to go with the oneshot approach.